pkg/controller/cloud/node_lifecycle_controller.go

```diff
@@ -131,6 +132,21 @@ func (c *CloudNodeLifecycleController) MonitorNodes() {
 	}
 
 	for _, node := range nodes {
+		// Check whether the node "unmanaged" annotation value is explicitly set to true, in which case the node is skipped.
+		// Otherwise, the node must be considered, and we proceed to checking whether we must delete it or apply the shutdown taint.
+		if value, exists := node.Annotations[UnmanagedNodeAnnotationKey]; exists {
```

Comment:
Prefer to base this on existence of the key only. Makes the usability of the annotation and the code cleaner. What do you think?

Reply:
I'm pretty fine with that. Can we get one more +1 from an approver?

Reply:
yeah, let's see what @cheftako has to say :)

Reply:
My 0.02 is this should be a label to match how other features work like how we can [exclude a node from a load balancer](https://github.com/kubernetes/kubernetes/blob/8a325fdbd1e5b41daeaa4524a39df9423f61d701/pkg/controller/service/service_controller.go#L70). Also, it should be an existence check to match how other features work.

Reply:
Using a label selector also lets us do server side filtering to not return these instances to the CCM at all.